cli/context/store/store.go

```diff
@@ -18,15 +18,47 @@ import (
 
 // Store provides a context store for easily remembering endpoints configuration
 type Store interface {
+	Reader
+	Lister
+	Writer
+	StorageInfo
+}
+
+// Reader provides read-only (without list) access to context data
+type Reader interface {
+	GetContextMetadata(name string) (ContextMetadata, error)
+	ListContextTLSFiles(name string) (map[string]EndpointFiles, error)
+	GetContextTLSData(contextName, endpointName, fileName string) ([]byte, error)
```

Comment:
It might be too late (not sure), but I keep looking at these and think: we should remove `Context` from these names.

I did a quick find/replace to see what it would look like, and pushed it to a branch; https://github.com/thaJeztah/cli/tree/remove_context_from_names

(it's based on this PR, so look at the last commit; https://github.com/docker/cli/commit/1068242c1545a1a4ac0c7a33a7e6c36d96b6b329)

Reply:
I aggree with this one, and if we want to do it, we have to do it before 19.03 is out I think. It will have impact on docker-app code, and on my current PR in UCP for adding a context file in the bundle though. I'll do it in a followup PR, so that we can choose to merge it or not.